src/os_crypto/shared/keys.c

```diff
@@ -217,12 +217,8 @@ void OS_ReadKeys(keystore *keys, int rehash_keys, int save_removed, int no_limit
     __memclear(id, name, ip, key, KEYSIZE + 1);
     memset(buffer, '\0', OS_BUFFER_SIZE + 1);
 
-    if (!fp) {
-        return;
-    }
-
     /* Read each line. Lines are divided as "id name ip key" */
-    while (fgets(buffer, OS_BUFFER_SIZE, fp) != NULL) {
+    while (fp && fgets(buffer, OS_BUFFER_SIZE, fp) != NULL) {
```

Comment:
I recomend to use a unique if(fp) block that covers the while block and the fclose(). As fp is not intended to be modified inside while it will be more clear to read, and avoid multi checking.

Reply:
Agreed